src/store/GlobalState.tsx

```diff
@@ -0,0 +1,44 @@
+import { createContext, useEffect, useState } from 'react';
+import { collection, query, onSnapshot } from 'firebase/firestore';
+
+import db from '../firebase';
+
+import { Expense } from '../expense';
+
+interface GlobalState {
+  expenses: Expense[];
+}
+
+const initialState: GlobalState = { expenses: [] };
+
+export const GlobalContext = createContext(initialState);
+
+export const GlobalProvider = ({ children }) => {
+  const [state, setState] = useState(initialState);
+
+  // Listen for changes to any doc in "expenses" collection and update expenses in app
+  useEffect(() => {
+    try {
+      const q = query(collection(db, 'expenses'));
+
+      onSnapshot(q, (querySnapshot) => {
+        const expenses: Expense[] = [];
```

Comment:
this logic seems like it belongs into the expense class (e.g. `fetchAllExpenses`)